unknown option: -soname on Mac OS X

When run `make` I get an error:
```
cc -Wl,-soname,libhoedown.so.3 -shared src/autolink.o src/buffer.o src/document.o src/escape.o src/html.o src/html_blocks.o src/html_smartypants.o src/stack.o src/version.o  -o libhoedown.so.3
ld: unknown option: -soname
```
After replacing it with `install_name` I get only 2 warnings:
```
bin/hoedown.c:446:10: warning: comparison of integers of different signs: 'clock_t' (aka 'unsigned long') and 'int' [-Wsign-compare]
                if (t1 == -1 || t2 == -1) {
                    ~~ ^  ~~
bin/hoedown.c:446:22: warning: comparison of integers of different signs: 'clock_t' (aka 'unsigned long') and 'int' [-Wsign-compare]
                if (t1 == -1 || t2 == -1) {
```
Mb should add it to README?

--- a/Makefile
+++ b/Makefile
@@ -9,6 +9,11 @@ ifneq ($(OS),Windows_NT)
 	HOEDOWN_CFLAGS += -fPIC
 endif
 
+SONAME = -soname
+ifeq ($(shell uname -s),Darwin)
+	SONAME = -install_name
+endif
+
 HOEDOWN_SRC=\
 	src/autolink.o \
 	src/buffer.o \
@@ -30,7 +35,7 @@ libhoedown.so: libhoedown.so.3
 	ln -f -s $^ $@
 
 libhoedown.so.3: $(HOEDOWN_SRC)
-	$(CC) -Wl,-soname,$(@F) -shared $^ $(LDFLAGS) -o $@
+	$(CC) -Wl,$(SONAME),$(@F) -shared $^ $(LDFLAGS) -o $@
 
 libhoedown.a: $(HOEDOWN_SRC)
 	$(AR) rcs libhoedown.a $^

